src/System.Runtime/ref/System.Runtime.cs

```diff
@@ -7211,6 +7222,9 @@ public sealed partial class SerializationInfo
         public ulong GetUInt64(string name) { throw null; }
         public object GetValue(string name, System.Type type) { throw null; }
         public void SetType(System.Type type) { }
+        public static DeserializationToken StartDeserialization() { throw null; }
+        public static void ThrowIfDeserializationInProgress() { }
+        public static void ThrowIfDeserializationInProgress(string switchSuffix, ref int cachedValue) { }
```

Comment:
I don't remember this API being reviewed and validated.  It's very strange to me.  Why can't a consumer just use AppContext directly?  Seems like we can count on one or two hands the number of call sites we'd ever have to this method, and they can actually do it more efficiently than this method is, e.g. this method is doing string validation, string concatenation, validation on the ref cachedValue, etc., none of which the call site would need to do.

Reply:
From the [notes](https://github.com/dotnet/corefx/issues/34555#issuecomment-455693313) 
> Provide throw helper with switch

As you say, upstack callers could skip the convenience method if they prefer to avoid the validation. (The expensive stuff only happens once or if it's about to throw though).